### PR TITLE
Remove Netif name and rely on enumerated Interface Id's for netif.

### PR DESCRIPTION
**Pull request by pvanhorn**
A defined Netif Interface ID enumeration has been added to openthread-types.h, allowing code both inside and out of OpenThread to specify well known InterfaceId values, such as within a otMessageInfo struct.

The Name member and associated API's, has been removed from the netif. This feature was not used and is no longer necessary now that all allowable Interface ID's are exposed in openthread-types.h

### CODE REVIEW

File: src/core/thread/topology.hpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ class Neighbor
      */
     enum State
 #if _WIN32
-        : unsigned int
```
Comment (pvanhorn) [inline src/core/thread/topology.hpp]: Unrelated to commit, make pretty just removed a tab.

**Comment by pvanhorn** [comment]
someone needs to explain to me why when I run make pretty at my desk it changes topology.hpp. But when travis runs make pretty it fails at topology.hpp?

**Comment by jwhui** [comment]
@pvanhorn, not sure why you are seeing different results.  What version of astyle do you have?

FWIW, I tried `make pretty` locally and see the same behavior as Travis.

My version:

```bash
$ astyle --version
Artistic Style Version 2.05.1
```
